packages/client/Atmosphere.ts

```diff
@@ -115,17 +123,31 @@ export default class Atmosphere extends Environment {
     })
   }
 
-  fetchHTTP = async (body: object, connectionId?: string) => {
-    const res = await fetch('/graphql', {
+  fetchHTTP = async (body: FetchHTTPData, connectionId?: string) => {
+    const req = {
       method: 'POST',
       headers: {
         accept: 'application/json',
         Authorization: this.authToken ? `Bearer ${this.authToken}` : '',
-        'content-type': 'application/json',
         'x-correlation-id': connectionId || ''
       },
-      body: JSON.stringify(body)
-    })
+    }
+    const uploadables = body.payload.variables?.uploadables
+    delete body.payload.variables?.uploadables
+    if (uploadables) {
+      if (!window.FormData) throw new Error('Uploading files without `FormData` not supported')
+      const formData = new FormData()
+      formData.append('body', JSON.stringify(body))
+      Object.keys(uploadables).forEach(key => {
+        if (!Object.prototype.hasOwnProperty.call(uploadables, key)) return
+        formData.append(key, uploadables[key])
```

Comment:
resolved via #4346

Reply:
hey @mattkrick , after attempting to work with your changes in #4346 , I think there may have been miscommunication. the way the code was written originally, there was expected to be only two fields: (1) `"body"`; and (2) `"file"`.  "body" contains a json stringified version of `StartMessage` containing `OperationPayload`. The only way an uploadable could overwrite anything was if the uploadable was named "body".

in my most recent changes, "body" is kept as-is, but each uploadable is prefixed with "uploadables" key. An uploadable can be named anything. E.g. `uploadables.file` or `uploadables.image`. This way, there's no way an uploadable can overwrite another field when constructing `formData`.*

i prefer this way because we avoid having to manually deserialize and reconstruct the request message. 

Curious to know your thoughts and if you still have any concerns.

*(Later in `parseFormBody`, the uploadable overwrites its same key name in `body.payload.variables`, but this is by design, so we can simply access the uploadable as a variable in the mutation resolver.)